### PR TITLE
Add vet to tests

### PR DESCRIPTION
**Pull request by liztio**
Our tests should vet, lint, and otherwise verify the hygiene of our code before we allow it into master

### CODE REVIEW

**Comment by liztio** [comment]
Fixes #124

File: pkg/config/loader_test.go [hunk 1]
```diff
@@ -145,7 +145,7 @@ func TestLoadAllPlugins(t *testing.T) {
 	}
 
 	if len(dsplugin.GetPodSpec().Containers) != 2 {
-		t.Fatalf("JobPlugin should have 1 container, got 2", len(jobplugin.GetPodSpec().Containers))
+		t.Fatalf("JobPlugin should have 2 containers, got %d", len(jobplugin.GetPodSpec().Containers))
```
Comment (timothysc) [inline pkg/config/loader_test.go]: 1 container, got %d

File: Makefile [hunk 2]
```diff
@@ -38,15 +38,21 @@ BUILDCMD = go build -o $(TARGET) -v -ldflags "-X github.com/heptio/sonobuoy/pkg/
...
 TESTARGS ?= -v -timeout 60s
-TEST = go test $(TEST_PKGS) $(TESTARGS)
 TEST_PKGS ?= $(GOTARGET)/cmd/... $(GOTARGET)/pkg/...
+TEST = go test $(TEST_PKGS) $(TESTARGS)
+
+
+# Vendor this someday
+INSTALL_GOLINT = go get -u github.com/golang/lint/golint
+GOLINT_FLAGS ?= -set_exit_status
+VET = go vet $(TEST_PKGS) && $(INSTALL_GOLINT) && golint $(GOLINT_FLAGS) $(TEST_PKGS)
```
Comment (timothysc) [inline Makefile]: You could probably break this into 2 var.. 
VET 
LINT

File: Makefile [hunk 3]
```diff
@@ -56,10 +62,13 @@ DOCKER_BUILD ?= $(DOCKER) run --rm -v $(DIR):$(BUILDMNT) -w $(BUILDMNT) $(BUILD_
...
-test: cbuild
-	 $(DOCKER_BUILD) '$(TEST)'
+test: cbuild vet
+	$(DOCKER_BUILD) '$(TEST)'
+
+vet:
+	$(DOCKER_BUILD) '$(VET)'
 
-container: test
+container: test vet
```
Comment (timothysc) [inline Makefile]: why is vet needed here?  

test: cbuild vet

File: Makefile (hunk 2 above)
Comment (timothysc) [reply Makefile]: Could also add a gfmt check but now were entering into the whole we should break up the target b/c we're chaining too many things.

File: pkg/config/loader_test.go (hunk 1 above)
Comment (liztio) [reply pkg/config/loader_test.go]: then why's it != 2?

Comment (timothysc) [reply pkg/config/loader_test.go]: wait now I need to dblcheck

Comment (timothysc) [reply pkg/config/loader_test.go]: my bad, ignore this comment.

**Pull request merged**